# 2018-10-19 summary

<!-- MarkdownTOC -->

- [Allele frequency methods](#allele-frequency-methods)
- [Heterozygosity](#heterozygosity)
      - [Overall heterozygosity](#overall-heterozygosity)
      - [Heterozygosity quantiles](#heterozygosity-quantiles)
  - [Popoolation heterozygosity estimates](#popoolation-heterozygosity-estimates)
- [DAPC](#dapc)
  - [Gene Expression](#gene-expression)
    - [generations together](#generations-together)
    - [each generation separately:](#each-generation-separately)
  - [SNPs](#snps)
  - [SNP results due to highly expressed genes?](#snp-results-due-to-highly-expressed-genes)
- [Convergence towards optimum](#convergence-towards-optimum)
  - [gene expression](#gene-expression-1)
    - [Treatment](#treatment)
    - [Line](#line)
  - [SNPs](#snps-1)
- [Functional Enrichment](#functional-enrichment)

<!-- /MarkdownTOC -->

All data are based on RNAseq from the F1, F2, F3 generations of all lines. 

Below I'll outline both the transcriptomics and the SNP analysis and results. 

## Allele frequency methods

Variants called with [varscan](http://dkoboldt.github.io/varscan/)
  
  1. All samps have at least 30x coverage
  2. where we call at least 4 replicates variable.
  3. max coverage per line: Median coverage is 132, remove those > 3*132=396x per sample
  4. This leaves 584,229 variable sites
  5. 10,986 genes remain (out of 45,994) 

## Heterozygosity

This was done two ways:

1. across each gene calculated as:`Hp <- (2 * sum(numMajor) * sum(numMinor)) / ((sum(numMajor) + sum(numMinor))^2)`

2. across each gene using popoolation. Mostly as a sanity check.

##### Overall heterozygosity 

**observe a subtle reduction in heterozygosity for `HH in AA`**

<img src="https://github.com/rsbrennan/tonsa_reciprocal/blob/master/figures/heterozygosity.png" width="500">

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/het_means.png" width="500">

##### Heterozygosity quantiles

**Heterozygosity is reduced in HH lines by F3- particuarly for `HH in AA`**

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/het_quantiles.png" width="600">


```
> fit <- aov(Het_01 ~ Line*Treatment*Generation, data=het_mean)
> summary(fit)
                          Df    Sum Sq   Mean Sq F value  Pr(>F)
Line                       1 0.0001877 1.877e-04   9.102 0.00467 **
Treatment                  1 0.0000611 6.111e-05   2.963 0.09377 .
Generation                 2 0.0000858 4.291e-05   2.081 0.13960
Line:Treatment             1 0.0000000 2.000e-08   0.001 0.97424
Line:Generation            2 0.0002367 1.184e-04   5.739 0.00686 **
Treatment:Generation       2 0.0000158 7.920e-06   0.384 0.68401
Line:Treatment:Generation  2 0.0000373 1.867e-05   0.905 0.41355
Residuals                 36 0.0007425 2.062e-05


> fit <- aov(Het_05 ~ Line*Treatment*Generation, data=het_mean)
> summary(fit)
                          Df    Sum Sq   Mean Sq F value  Pr(>F)
Line                       1 0.0000950 9.500e-05   7.975 0.00768 **
Treatment                  1 0.0000523 5.231e-05   4.392 0.04321 *
Generation                 2 0.0000621 3.105e-05   2.606 0.08767 .
Line:Treatment             1 0.0000000 3.000e-08   0.003 0.95777
Line:Generation            2 0.0000754 3.771e-05   3.166 0.05413 .
Treatment:Generation       2 0.0000036 1.800e-06   0.151 0.86015
Line:Treatment:Generation  2 0.0000293 1.464e-05   1.229 0.30457
Residuals                 36 0.0004288 1.191e-05

```

If we classify groups by Line_Treatment, which I don't really think is justified, we get a bit more nuanced stats, where all significant comb:Generation interactions are driven by HH in AA. 

```
> fit <- aov(Het_01 ~ comb*Generation, data=het_mean)
> summary(fit)
                Df    Sum Sq   Mean Sq F value Pr(>F)
comb             3 0.0002489 8.295e-05   4.022 0.0145 *
Generation       2 0.0000858 4.291e-05   2.081 0.1396
comb:Generation  6 0.0002899 4.832e-05   2.343 0.0518 .
Residuals       36 0.0007425 2.062e-05

> out <- TukeyHSD(fit, "comb:Generation")$`comb:Generation`[,4]
> out[which(out < 0.05)]
AAHH:F3-HHAA:F1 AAHH:F3-HHHH:F1 AAHH:F3-HHAA:F2 AAHH:F3-AAAA:F3 HHAA:F3-AAHH:F3
    0.047151296     0.041222686     0.004372989     0.020953950     0.006496518

```

### Popoolation heterozygosity estimates

as a sanity check for above. Still working on this. 



## DAPC

These results, below, suggest that:

  1. Both AA and HH lines can alter gene expression patterns to begin to converge on the opposite line;
  2. HH in AA shows a shift in allele frequencies towards AA while AA in HH does not. This suggests that HH lines are maladapted to AA conditions and/or have lost plasticity that AA lines maintain
    - that is, AA line can alter expression without altering allele frequencies. HH cannot. 


### Gene Expression

Using DESeq2. Transcriptome wide expression differences were characterized using DAPC- minimizing within group variance (AAAA and HHHH), maximizing between group variance. 

Transcripts were filtered to remove any where count < 10 in more than 90% of samples. This leaves 18,746 when all generations considered together; ~23,000 when considered separately. 

Ran DAPC on 1. all gens and samples together. 2. Each generation separately.  
I think running each generation separately makes sense because this will control for variation in handling, etc, between sampling time. Regardless, both are presented below. 


#### generations together

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/dapc_RNA_all.png" width="350">

Test for differences in DAPC using MCMCglmm:

We can ask if the posterior distributions of the fixed effects of line:treatment are different by taking the difference between the posteriors of the effect of treatment in the HH and AA lines: 

All generations together:

```
HPDinterval(awayDelta)
          lower    upper
var1 -0.4417333 2.082827

```

The 95% credibile interval includes 0, and 92% of our posterior distribution suggests that the HH line has greater plasticity. This suggests there is not a difference in plasticity.

#### each generation separately:

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/dapc_RNA_gen.png" width="350">

```
F1:
         lower       upper
var1 -3.168696 -0.08401299

98% of posterior supports AA being more plastic.

F2:
         lower    upper
var1 -3.952902 1.711024

82% of posterior supports AA being more plastic.

F3:
         lower    upper
var1 -2.521452 1.640246
59% of posterior supports AA being more plastic.

```


### SNPs

**These results show that allele frequencies for `HH in AA` show a larger shift than `AA in HH`**

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/dapc_snps_all.png" width="350">

```

         lower    upper
var1 0.6870455 3.119645

0.999 of posterior supports greater shift in allele freqs in HH line

```

### SNP results due to highly expressed genes?

There is some risk that the differentially expressed genes are driving the allele frequency differences due to allele specific expression (or similar). Looking at the correlation of contribution of to the DF for the transcripts SNPs. In short, they are not correlated. This is good and suggests that expression is not driving allele freq differences. 

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/loading_corr.png" width="350">

## Convergence towards optimum

If we assume that AAAA and HHHH represent and optimum or adaptive phenotype/genotype, we can try to quantify the convergence towards this optimum in the treatment conditions. 

My basic thinking is very similar to [this oyster paper](https://doi.org/10.1038/s41559-018-0668-2): 

  1. We know that HHHH and AAAA have been in their respective environments for ~20 generations. We also know that these lines have recovered fitness to the point that we think they've adapted to their respective environments. Therefore, there differences in gene expression between these lines+treatments should represent the adaptive expression pattern. 
  2. We can ask how similar in expression each line+treatment combination is to the "adaptive" expression patterns.
  3. We can ask the same question for allele frequencies.  

### gene expression

**These results suggest that `AA in HH` and `HH in AA` are converging in expression towards their respective treatment optimum.**

Trying to figure out how lines/treatments converge on the adaptive gene expression. 

For example, our "adaptive" comparison is `[AA in AA]/[HH in HH]`, we can pull out the differentially expressed trascnripts between these two. We could then ask how similar the AA line is to this DGE. First, by looking at the consistent line effect, so `[AA in HH]/[HH in HH]` vs `[AA in AA]/[HH in HH]`. This is asking, "how similar is `[AA in AA]` expression to `AA expression in HH`". We can also look at the treatment effect, `[AA in HH]/[AA in AA]` vs `[HH in HH]/[AA in AA]`. These are very similar comparisons, just with a different perspective. 
 
So for each plot below, there is the log fold change of the comparison. x-axis is always the "adaptive" response `[AA in AA]/[HH in HH]`, or flipped. y-axis is the treatment or line effect. 

#### Treatment 

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/trt_DGE.png" width="600">

#### Line

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/line_DGE.png" width="600">

### SNPs

**These results suggest that `HH in AA` is converging in allele frequency towards `AA in AA`. `AA in HH` does not converge on `HH in HH`**

Plots below show the average change in allele frequency between groups. Basic concept is the same as described above. 

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/trt_allele_freq.png" width="600">


## Functional Enrichment

in progress. 


