# 2018-10-19 summary

All data are based on RNAseq from the F1, F2, F3 generations of all lines. 

Below I'll outline both the transcriptomics and the SNP analysis and results. 

## Transcriptomics methods





## Allele frequency methods

Variants called with [varscan](http://dkoboldt.github.io/varscan/)
  
  1. All samps have at least 30x coverage
  2. where we call at least 4 replicates variable.
  3. max coverage per line: Median coverage is 132, remove those > 3*132=396x per sample
  4. This leaves 584,229 variable sites
  5. 10,986 genes remain (out of 45,994) 

## Heterozygosity

This was done two ways:

1. across each gene calculated as:`Hp <- (2 * sum(numMajor) * sum(numMinor)) / ((sum(numMajor) + sum(numMinor))^2)`

2. across each gene using popoolation. Mostly as a sanity check.

##### Overall heterozygosity 

<img src="https://github.com/rsbrennan/tonsa_reciprocal/blob/master/figures/heterozygosity.png" width="500">

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/het_means.png" width="500">

##### Heterozygosity quantiles

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_f20/master/figures/het_quantiles.png" width="500">


```
> fit <- aov(Het_01 ~ Line*Treatment*Generation, data=het_mean)
> summary(fit)
                          Df    Sum Sq   Mean Sq F value  Pr(>F)
Line                       1 0.0001877 1.877e-04   9.102 0.00467 **
Treatment                  1 0.0000611 6.111e-05   2.963 0.09377 .
Generation                 2 0.0000858 4.291e-05   2.081 0.13960
Line:Treatment             1 0.0000000 2.000e-08   0.001 0.97424
Line:Generation            2 0.0002367 1.184e-04   5.739 0.00686 **
Treatment:Generation       2 0.0000158 7.920e-06   0.384 0.68401
Line:Treatment:Generation  2 0.0000373 1.867e-05   0.905 0.41355
Residuals                 36 0.0007425 2.062e-05


> fit <- aov(Het_05 ~ Line*Treatment*Generation, data=het_mean)
> summary(fit)
                          Df    Sum Sq   Mean Sq F value  Pr(>F)
Line                       1 0.0000950 9.500e-05   7.975 0.00768 **
Treatment                  1 0.0000523 5.231e-05   4.392 0.04321 *
Generation                 2 0.0000621 3.105e-05   2.606 0.08767 .
Line:Treatment             1 0.0000000 3.000e-08   0.003 0.95777
Line:Generation            2 0.0000754 3.771e-05   3.166 0.05413 .
Treatment:Generation       2 0.0000036 1.800e-06   0.151 0.86015
Line:Treatment:Generation  2 0.0000293 1.464e-05   1.229 0.30457
Residuals                 36 0.0004288 1.191e-05

```

If we classify groups by Line_Treatment, which I don't really think is justified, we get a bit more nuanced stats, where all significant comb:Generation interactions are driven by HH in AA. 

```
> fit <- aov(Het_01 ~ comb*Generation, data=het_mean)
> summary(fit)
                Df    Sum Sq   Mean Sq F value Pr(>F)
comb             3 0.0002489 8.295e-05   4.022 0.0145 *
Generation       2 0.0000858 4.291e-05   2.081 0.1396
comb:Generation  6 0.0002899 4.832e-05   2.343 0.0518 .
Residuals       36 0.0007425 2.062e-05

> out <- TukeyHSD(fit, "comb:Generation")$`comb:Generation`[,4]
> out[which(out < 0.05)]
AAHH:F3-HHAA:F1 AAHH:F3-HHHH:F1 AAHH:F3-HHAA:F2 AAHH:F3-AAAA:F3 HHAA:F3-AAHH:F3
    0.047151296     0.041222686     0.004372989     0.020953950     0.006496518

```

## Gene Expression

Using DESeq2. Transcriptome wide expression differences were characterized using DAPC- minimizing within group variance (AAAA and HHHH), maximizing between group variance. 

Transcripts were filtered to remove any where count < 10 in more than 90% of samples. This leaves 18,746 when all generations considered together; ~23,000 when considered separately. 

Ran DAPC on 1. all gens and samples together. 2. Each generation separately.  
I think running each generation separately makes sense because this will control for variation in handling, etc, between sampling time. Regardless, both are presented below. 


<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/dapc_RNA_all.png" width="500">

Test for differences in DAPC using MCMCglmm:

We can ask if the posterior distributions of the fixed effects of line:treatment are different by taking the difference between the posteriors of the effect of treatment in the HH and AA lines: 

All generations together:

```
HPDinterval(awayDelta)
          lower    upper
var1 -0.4417333 2.082827

```

The 95% credibile interval includes 0, and 92% of our posterior distribution suggests that the HH line has greater plasticity. 

DAPC for each generation separately:

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/dapc_RNA_gen.png" width="500">

```
F1:
         lower       upper
var1 -3.168696 -0.08401299

98% of posterior supports AA being more plastic.

F2:
         lower    upper
var1 -3.952902 1.711024

82% of posterior supports AA being more plastic.

F3:
         lower    upper
var1 -2.521452 1.640246
59% of posterior supports AA being more plastic.

```


## SNPs

```

         lower    upper
var1 0.6870455 3.119645

0.0011 of posterior supports greater shift in allele freqs in HH line

```

<img src="https://raw.githubusercontent.com/rsbrennan/tonsa_reciprocal/master/figures/dapc_snps_all.png" width="500">











